src/utils/DateUtils.ts

```diff
@@ -270,28 +270,28 @@ export class DateUtils {
     let correctedFormat = format;
 
     const fourLowercaseY = DateUtils.buildRegexMatchingExactCharSequence('y', 4);
-    correctedFormat = correctedFormat.replace(fourLowercaseY, 'YYYY');
+    correctedFormat = correctedFormat.replace(fourLowercaseY, '$1YYYY');
 
     const twoLowercaseY = DateUtils.buildRegexMatchingExactCharSequence('y', 2);
-    correctedFormat = correctedFormat.replace(twoLowercaseY, 'YY');
+    correctedFormat = correctedFormat.replace(twoLowercaseY, '$1YY');
 
     const twoLowercaseD = DateUtils.buildRegexMatchingExactCharSequence('d', 2);
-    correctedFormat = correctedFormat.replace(twoLowercaseD, 'DD');
+    correctedFormat = correctedFormat.replace(twoLowercaseD, '$1DD');
 
     const oneLowercaseD = DateUtils.buildRegexMatchingExactCharSequence('d', 1);
-    correctedFormat = correctedFormat.replace(oneLowercaseD, 'D');
+    correctedFormat = correctedFormat.replace(oneLowercaseD, '$1D');
 
     const twoLowercaseH = DateUtils.buildRegexMatchingExactCharSequence('h', 2);
-    correctedFormat = correctedFormat.replace(twoLowercaseH, 'H');
+    correctedFormat = correctedFormat.replace(twoLowercaseH, '$1H');
 
     return correctedFormat;
   }
 
   private static buildRegexMatchingExactCharSequence(char: string, sequenceLength: number) {
-    const negativeLookBehind = `(?<!${char})`;
+    const negativeNonCapturingGroup = `(?:([^${char}]|^))`; // look-behind is not supported in Firefox
     const charSequence = `${char}{${sequenceLength}}`;
     const negativeLookAhead = `(?!${char})`;
```

Comment:
Do we know if the negative LookAhead work?

Reply:
Yes, both negative and positive lookaheads are supported in javascript across the common browsers.